app/main.py

```diff
@@ -1,6 +1,20 @@
 from typing import Callable
+import functools
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_results = {}
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs) -> dict:
```

Comment:
Please change the return annotation from dict to Any, as we don't know the type of the return result